mlrun/runtimes/local.py

```diff
@@ -395,6 +395,8 @@ def read_stderr(stderr):
         stderr_consumer_thread.join()
         err = stderr.getvalue()
 
+    process.wait()
```

Comment:
This fix is needed here because this broke in CI, triggered by a new warning due to pyarrow calling `is_sparse`, which is deprecated in pandas 2.

Reply:
wouldnt that block the process now until it finishes?

Reply:
That's what this function is supposed to do. In fact it already does it by waiting for stderr, but there's a race condition there that causes the return code to be `None`.